storage/pebble/registers.go

```diff
@@ -61,7 +69,14 @@ func (s *Registers) Store(
 	height uint64,
 	entries flow.RegisterEntries,
 ) error {
-	//  TODO: add value check once LatestHeight is implemented
+	// value check
+	firstHeight, latestHeight, err := s.getHeightRange()
+	if err != nil {
+		return fmt.Errorf("unable to perform height validation: %w", err)
+	}
+	if height > latestHeight+1 || height < firstHeight {
+		return storage.ErrHeightNotIndexed
+	}
```

Comment:
Please refer to [the behavior described here](https://github.com/onflow/flow-go/pull/4656/files#diff-921bdc69d712e22e7fe6c2ee7c2761c4c68e9c2f39ba73a2031eb79361c8f330R26-R28)

- Store should not return `ErrHeightNotIndexed` error, which is only returned by Get method
- the only accepted height is either latest height or latest height + 1.